internal/k8s/loader/load.go

```diff
@@ -64,7 +64,8 @@ func FetchClusterLogForwarder(k8sClient client.Client, namespace, name string, i
 	forwarder.Spec, extras, migrationMessages = migrations.MigrateClusterLogForwarder(namespace, name, forwarder.Spec, fetchClusterLogging().Spec.LogStore, extras, internalLogStoreSecret, saTokenSecret)
 	setMigrationStatusConditions(&forwarder.Status, migrationMessages)
 
-	extras[constants.ClusterLoggingAvailable] = (fetchClusterLogging().Name != "")
+	clusterLogging := fetchClusterLogging()
+	extras[constants.ClusterLoggingAvailable] = (clusterLogging.Name != "" && clusterLogging.Spec.Collection != nil)
```

Comment:
I can't wrap my head around this exactly but my expectation is:

* ns:openshift-logging, name:instance - Require collection stanza to be provided by admin
* ns:other, name:other - Optional collection stanza to be provided by admin

The general flow is: load, migrate, validate.  I believe this change still allows an admin to deploy a forwarder without explicitly specifying a ClusterLogging object but I want to be certain.  I believe we have a test to verify and if we don't I would expect this PR to add one.

Reply:
> The general flow is: load, migrate, validate. I believe this change still allows an admin to deploy a forwarder without explicitly specifying a ClusterLogging object but I want to be certain.

When I create a ClusterLogForwarder without a ClusterLogging object (in my tests), the operator doesn't deploy any resources. I'll dig into this, but given that my exposure to / knowledge of the operator is a bit limited .. What's the expected outcome of just deploying the ClusterLogForwarder resource, without a ClusterLogging object?

Reply:
I'll definitely add an E2E test for this

Reply:
> When I create a ClusterLogForwarder without a ClusterLogging object (in my tests), the operator doesn't deploy any resources

This should only be the case for when ClusterLogForwarder(ns:openshift,name:instance) exists with no similarly matched NS/name ClusterLogging instances which is the legacy case. All other combinations CLF name/ns are supposed to deploy a collector without needing a ClusterLogging to define the collector. That is realized by [this](https://github.com/openshift/cluster-logging-operator/blob/master/controllers/forwarding/forwarding_controller.go#L147-L158) code.  This [e2e](https://github.com/openshift/cluster-logging-operator/blob/master/hack/testing-olm/test-1343-minimal-multi-clf.sh#L64) test should verify that behavior

Reply:
Alright, got it:
```
LOG_LEVEL=10 make run NAMESPACE=""
```

```
#!/bin/bash

cat <<'EOF' | oc apply -f -
apiVersion: v1
kind: Namespace
metadata:
  name: log-forwarder
EOF

oc create sa log-forwarder -n log-forwarder
oc adm policy add-cluster-role-to-user collect-application-logs system:serviceaccount:log-forwarder:log-forwarder
oc adm policy add-cluster-role-to-user collect-infrastructure-logs system:serviceaccount:log-forwarder:log-forwarder
oc adm policy add-cluster-role-to-user collect-audit-logs system:serviceaccount:log-forwarder:log-forwarder

cat <<'EOF' | oc apply -f -
apiVersion: logging.openshift.io/v1
kind: ClusterLogForwarder
metadata:
  name: log-forwarder
  namespace: log-forwarder
spec:
  serviceAccountName: log-forwarder
  pipelines:
   - inputRefs:
     - audit
     outputRefs:
     - loki-external
     labels:
       job: openshift-audit
       clusterid: sno10
   - inputRefs:
     - application
     outputRefs:
     - loki-external
     labels:
       job: openshift-application
       clusterid: sno10
   - inputRefs:
     - infrastructure
     outputRefs:
     - loki-external
     labels:
       job: openshift-infrastructure
       clusterid: sno10
  outputs:
  - name: loki-external
    type: loki
    url: http://192.168.18.75:3100
EOF
```

That still deploys the forwarder.

Reply:
Interesting, so all 3 issues that I found affect the legacy implementation, including the infinite loop for the status